### PR TITLE
Add minikube/releases/stable.txt for downloading

### PR DESCRIPTION
**Pull request by r2d4**
The latest stable kubectl can be used for downloading with the script

```
curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
```

We should allow minikube to be downloaded in a similar manner.  Right
now, downstream docs and demos need to hardcode the version.  This adds
gs://minikube/releases/stable.txt

With this PR, you will be able to reference

```
curl -LO https://storage.googleapis.com/minikube/releases/$(curl -s https://storage.googleapis.com/minikube/releases/stable.txt)/minikube-linux-amd64
```

It also changes the notification update on new releases to just parse
this file instead of the releases.json, as it will be much more
straightforward.

### CODE REVIEW

File: pkg/minikube/notify/notify.go [hunk 1]
```diff
@@ -82,41 +81,17 @@ func shouldCheckURLVersion(filePath string) bool {
...
-}
-
-type Releases []Release
-
-func getJson(url string, target *Releases) error {
+func getLatestVersionFromURL(url string) (semver.Version, error) {
 	r, err := http.Get(url)
 	if err != nil {
-		return errors.Wrap(err, "Error getting minikube version url via http")
+		return semver.Version{}, err
```
Comment (aaron-prindle) [inline pkg/minikube/notify/notify.go]: Should this remain errors.Wrap?  This is actually an issue all over in that we have some errors which are wrapped to have stack tracing and some which are not.

**Comment by dlorenc** [comment]
Don't we already publish to the latest directory too?

**Comment by r2d4** [comment]
Oops I didn't see that we also publish to latest.  That's probably a much better idea.

**Comment by dlorenc** [comment]
> Oops I didn't see that we also publish to latest. That's probably a much better idea.

I'm fine adding this too, if it's a convention other k8s projects use. This feels a little simpler though.

**Comment by r2d4** [comment]
Yeah i'd rather not maintain another way to do this.  release/latest/ is simple enough.

Closing this

**Pull request closed without merging**